### PR TITLE
Add Heroku deployment hint to README

### PR DESCRIPTION
**Pull request by cseelus**
### Summary

As mentioned in issue #249

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -832,6 +832,8 @@ This behavior is optional & can be disabled by either setting an `WEBPACKER_PREC
 
 When compiling assets for production on a remote server, such as a continuous integration environment, it's recommended to use `yarn install --frozen-lockfile` to install NPM packages on the remote host to ensure that the installed packages match the `yarn.lock` file.
 
+Please note that on Heroku, the `heroku/nodejs` buildpack needs to be included *before* the `heroku/ruby` buildpack for precompilation to run successfully.
```
Comment (justin808) [inline README.md:835]: Awesome, can we get a note regarding docker builds as well?

Comment (justin808) [reply README.md:835]: Your production build process is responsible for running `yarn install` before `rake assets:precompile`. For example, if you are on Heroku, the `heroku/nodejs` buildpack must run **prior** to the `heroku/ruby` buildpack for precompilation to run successfully.

Comment (ahangarha) [reply README.md:835]: I think the better place to add this is in [docs/deployment.md](https://github.com/shakacode/shakapacker/blob/master/docs/deployment.md).

Comment (cseelus) [reply README.md]: I've moved it to `docs/deployment.md`.

Where should I add the note regarding docker builds?

Comment (cseelus) [reply README.md]: What should I add re docker builds, or is the text from @justin808 that I added sufficient?

For the CHANGELOG

Should I add it to `## [v6.5.6] - February 11, 2023`:

- Please note, that your production build process is responsible for running `yarn install` before `rake assets:precompile`. See `docs/deployment.md` for more info.

**Comment by justin808** [comment]
Thanks @cseelus!

**Pull request merged**